src/canopy_calcs.F90

```diff
@@ -209,123 +219,168 @@ SUBROUTINE canopy_calcs
                                 end if
                             end if
 
+!.......user option to calculate in-canopy leafage influence and assigning LAI as per timestep
+
+
+                            if (leafage_opt .eq. 0) then
+                                ! Initialize pastlai and currentlai based on current timestep
+                                if (nn .eq. 1) then
+                                    currentlai = lairef
+                                    pastlai = currentlai
+                                else
+                                    pastlai = currentlai
+                                    currentlai = lairef
+                                end if
+
+                                ! Print diagnostics for currentlai and pastlai at each timestep nn
+                                print *, 'Timestep (nn):', nn, 'Current LAI:', currentlai, 'Past LAI:', pastlai
+
+                                ! Check if the lai_tstep is greater than time_intvl
+                                if (lai_tstep .ge. time_intvl) then
+                                    tsteplai = lai_tstep/86400.0_rk
+                                else
+                                    WRITE (*, *) "Error: Input LAI time step cannot be less than model time step...exiting!!!"
```

Comment:
How come?

Reply:
@drnimbusrain Think this was one of your suggestions earlier:   https://github.com/noaa-oar-arl/canopy-app/pull/92#discussion_r1346638763

Reply:
Ah, sorry, that's right.  We can leave it for now.